File: 1 Two Sum/soln.go
func twoSum(nums []int, target int) []int {
    
    m := make(map[int]int)
    
    for i := 0; i < len(nums); i++{
        v := target - nums[i]
        if _, ok := m[v]; ok{
            return []int{i, m[v]}
        } else{
            m[nums[i]] = i;
        }
    }
    return []int{0, 0}
}

File: 1 Two Sum/soln.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> m;
        
        for (int i = 0; i < nums.size(); i++){
            int diff = target - nums[i];
            
            if (m.find(diff) != m.end()){
                return {m[diff], i};
            }
            m[nums[i]] = i;
        }
        return {};
    }
};